controllers/kuadrant_status.go

```diff
@@ -119,29 +119,23 @@ func (r *KuadrantReconciler) readyCondition(ctx context.Context, kObj *kuadrantv
 	return cond, nil
 }
 
-func (r *KuadrantReconciler) checkLimitadorAvailable(ctx context.Context, kObj *kuadrantv1beta1.Kuadrant) (*string, error) {
-	// Should be implemented reading the Limitador CR's status conditions.
-	// Not implemented yet in the limitador's operator
-	deployment := &appsv1.Deployment{}
-	dKey := client.ObjectKey{Name: "limitador", Namespace: kObj.Namespace}
-	err := r.Client().Get(ctx, dKey, deployment)
+func (r *KuadrantReconciler) checkLimitadorReady(ctx context.Context, kObj *kuadrantv1beta1.Kuadrant) (*string, error) {
+	limitadorObj := &limitadorv1alpha1.Limitador{}
+	limitadorKey := client.ObjectKey{Name: common.LimitadorName, Namespace: kObj.Namespace}
+
+	err := r.Client().Get(ctx, limitadorKey, limitadorObj)
 	if err != nil && !errors.IsNotFound(err) {
 		return nil, err
 	}
 
-	if err != nil && errors.IsNotFound(err) {
-		tmp := err.Error()
-		return &tmp, nil
-	}
-
-	availableCondition := common.FindDeploymentStatusCondition(deployment.Status.Conditions, "Available")
-	if availableCondition == nil {
-		tmp := "Available condition not found"
-		return &tmp, nil
+	if errors.IsNotFound(err) {
+		reason := "Limitador not found"
+		return &reason, nil
 	}
 
-	if availableCondition.Status != corev1.ConditionTrue {
-		return &availableCondition.Message, nil
+	statusConditionReady := meta.FindStatusCondition(limitadorObj.Status.Conditions, "Ready")
+	if statusConditionReady.Status != metav1.ConditionTrue {
+		return &statusConditionReady.Message, fmt.Errorf(statusConditionReady.Message)
```

Comment:
it should not return error. It is not an error to have the limitador not ready. It is a condition or state.